lib/matplotlib/scale.py

```diff
@@ -725,7 +776,12 @@ def scale_factory(scale, axis, **kwargs):
     axis : `~matplotlib.axis.Axis`
     """
     scale_cls = _api.check_getitem(_scale_mapping, scale=scale)
-    return scale_cls(axis, **kwargs)
+    try:
+        return scale_cls(axis, **kwargs)
+    except TypeError as e:
+        if 'unexpected keyword argument' in str(e) or 'positional argument' in str(e):
+            return scale_cls(**kwargs)
+        raise
```

Comment:
Are you sure this is safe? Basing logic on the exception message only can be brittle. Minimally, I'd want an explicit comment on what we catch here. But I'd rather split this out into a separte PR as we need tests.